components/net_traits/bluetooth_scanfilter.rs

```diff
@@ -30,36 +24,44 @@ pub struct BluetoothScanfilter {
 }
 
 impl BluetoothScanfilter {
-    fn is_empty_or_invalid_filter(&self) -> bool {
-        self.name.is_empty() &&
-        self.name_prefix.is_empty() &&
-        self.services.is_empty() &&
-        self.name.len() > MAX_NAME_LENGTH &&
-        self.name_prefix.len() > MAX_NAME_LENGTH
-    }
-
     pub fn new(name: String, name_prefix: String, services: Vec<String>) -> BluetoothScanfilter {
         BluetoothScanfilter {
             name: name,
             name_prefix: name_prefix,
             services: ServiceUUIDSequence::new(services),
         }
     }
+
+    pub fn get_name(&self) -> String {
+        self.name.clone()
+    }
+
+    pub fn get_name_prefix(&self) -> String {
+        self.name_prefix.clone()
+    }
+
+    pub fn get_services(&self) -> Vec<String> {
+        self.services.0.clone()
+    }
+
+    pub fn is_empty_or_invalid_filter(&self) -> bool {
+        (self.name.is_empty() && self.name_prefix.is_empty() && self.get_services().is_empty()) ||
+        self.name.len() > MAX_NAME_LENGTH ||
+        self.name_prefix.len() > MAX_NAME_LENGTH
+    }
 }
 
 #[derive(Deserialize, Serialize)]
-pub struct BluetoothScanfilterSequence(Vec<BluetoothScanfilter>);
+pub struct BluetoothScanfilterSequence(pub Vec<BluetoothScanfilter>);
```

Comment:
Because we need to use the inner Vec with a ```.0``` in another file https://github.com/szeged/servo/pull/21/files#diff-438a3422f5ba8c39dabd8e4e18928c51R100